zilliqa/src/api/eth.rs

```diff
@@ -816,6 +766,377 @@ fn get_uncle(_: Params, _: &Arc<Mutex<Node>>) -> Result<Option<String>> {
     Ok(None)
 }
 
+#[derive(Deserialize, Default)]
+#[serde(default, rename_all = "camelCase")]
+struct NewFilterParams {
+    from_block: Option<BlockNumberOrTag>,
+    to_block: Option<BlockNumberOrTag>,
+    address: Option<OneOrMany<Address>>,
+    topics: Vec<Option<OneOrMany<B256>>>,
+}
+
+fn new_filter(params: Params, node: &Arc<Mutex<Node>>) -> Result<String> {
+    let mut seq = params.sequence();
+    let mut params: NewFilterParams = seq.next()?;
+    expect_end_of_params(&mut seq, 1, 1)?;
+
+    let mut node = node.lock().unwrap();
+
+    if params.topics.len() > 4 {
+        // Too many topics provided, max of 4
+        return Err(anyhow!("Too many topics provided, maximum 4"));
+    }
+
+    if node.config.max_filters > 0 && node.filters.len() as u64 > node.config.max_filters {
+        return Err(anyhow!("Too many filters already registered"));
+    }
+
+    let mut topics = [None, None, None, None];
+    topics[..params.topics.len()].swap_with_slice(&mut params.topics[..]);
+
+    let from_block = params.from_block.unwrap_or(BlockNumberOrTag::Latest);
+    // let first_block = node.resolve_block_number(from_block)?.unwrap().number();
+
+    let first_block = match params.from_block.unwrap_or(BlockNumberOrTag::Latest) {
+        BlockNumberOrTag::Number(n) => n,
+        tag => node.resolve_block_number(tag)?.unwrap().number(),
+    };
+
+    let mut rng = rand::thread_rng();
+
+    let mut id = [0u8; 32];
```

Comment:
Do filter IDs really need to be 32 bytes long? What about just `rand::random::<u64>()`?

Reply:
This was what other Ethereum endpoints seemed to use for their filter ids